core/src/syscall/host.rs

```diff
@@ -123,6 +127,30 @@ impl Syscall for SyscallHostCall {
     let host = ctx.rt.host.as_deref_mut().expect("Missing host interface");
     let res = host.call(msg);
 
+    let output_size = if let Some(output) = res.output {
+      let mut output_ptr = ctx.rt.register(Register::X13);
+      let output_size = ctx.rt.register(Register::X14);
+      let output_size = min(output.len() as u32, output_size);
```

Comment:
Add a warning if `output.len() > output_size` so it's clear that output is being truncated